### PR TITLE
feat(sozo): automatically authorize models to migrated systems based on models specified in overlays

### PR DESCRIPTION
**Pull request by lambda-0x**
fix: #1819

### CODE REVIEW

File: crates/sozo/ops/src/migration/mod.rs [hunk 1]
```diff
@@ -139,7 +142,9 @@ where
 
         if let Some(migration_output) = migration_output {
             if !ws.config().offline() {
-                upload_metadata(ws, account, migration_output, txn_config).await?;
+                upload_metadata(ws, &account, migration_output.clone(), txn_config).await?;
+                let world = WorldContract::new(world_address, account);
+                auto_authorize(ws, &world, &txn_config, &local_manifest, &migration_output).await?;
```
Comment (glihm) [inline crates/sozo/ops/src/migration/mod.rs:147]: If this fails, we shouldn't abort the whole migration. Only a warning should be displayed to mention that the authorization transaction couldn't be send.

Comment (lambda-0x) [reply crates/sozo/ops/src/migration/mod.rs:147]: good catch ser, fixed it.

File: crates/saya/core/src/prover/serializer.rs
Comment (glihm) [inline crates/saya/core/src/prover/serializer.rs:1]: This file has been deleted. 👍

File: crates/sozo/ops/src/migration/auto_auth.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+pub fn compute_models_contracts(
+    ui: &Ui,
+    local_manifest: &BaseManifest,
+    migration_output: &MigrationOutput,
+) -> Result<Vec<crate::auth::ModelContract>> {
+    let mut res = vec![];
+    let local_contracts = &local_manifest.contracts;
+
+    // from all the contracts that where migrated successfully
```
Comment (glihm) [inline crates/sozo/ops/src/migration/auto_auth.rs:40]:
```suggestion
    // From all the contracts that were migrated successfully.
```
Trying to keep comment as punctuated sentences when possible.

File: crates/sozo/ops/src/migration/auto_auth.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            .iter()
+            .find(|c| migrated_contract.name == c.name)
+            .expect("we know this contract exists");
+
+        ui.print_sub(format!(
+            "Authorizing {} for Models: {:?}",
+            contract.name, contract.inner.writes
+        ));
+
+        // read all the models that its supposed to write to and
```
Comment (glihm) [inline crates/sozo/ops/src/migration/auto_auth.rs:53]:
```suggestion
        // read all the models that its supposed to write to ???
```

File: crates/sozo/ops/src/tests/migration.rs [hunk 4]
```diff
@@ -240,6 +245,65 @@ async fn migrate_with_metadata() {
...
+            .unwrap();
+
+    let overlay_manifest = OverlayManifest::load_from_path(
+        &manifest_base.join(MANIFESTS_DIR).join("dev").join(OVERLAYS_DIR),
+    )
+    .unwrap();
+
+    manifest.merge(overlay_manifest);
+
+    let sequencer =
```
Comment (glihm) [inline crates/sozo/ops/src/tests/migration.rs:268]: For the next PRs, once the Katana runner update is merged with Cairo 2.6, let's try to use Katana runner instead. For this one let's keep that. :)

**Review by glihm — APPROVED** [review]
Minor comments, rebase and good to go @lambda-0x! Thank for the great work here and this will be a nice improvement at devX level.

We need to finish the overlay documentation, I've started something but didn't take the time to wrap up and merge. Will do soon, sorry for the delay.

**Comment by glihm** [comment]
Let's follow up that with discussion for #1921. 👍

**Pull request merged**